pkg/gds/store/trtl/index.go

```diff
@@ -0,0 +1,364 @@
+package trtl
+
+import (
+	"context"
+	"io"
+
+	"github.com/rs/zerolog/log"
+	storeerrors "github.com/trisacrypto/directory/pkg/gds/store/errors"
+	"github.com/trisacrypto/directory/pkg/gds/store/index"
+	"github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+	"github.com/trisacrypto/directory/pkg/utils/wire"
+	gds "github.com/trisacrypto/trisa/pkg/trisa/gds/models/v1beta1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+//===========================================================================
+// Indexer
+//===========================================================================
+
+// Reindex rebuilds the name and country indices for the server and synchronizes them
+// back to disk to ensure they're complete and accurate.
+func (s *Store) Reindex() (err error) {
+	names := index.NewNamesIndex()
+	websites := index.NewWebsiteIndex()
+	countries := index.NewCountryIndex()
+	categories := index.NewCategoryIndex()
+
+	ctx, cancel := withContext(context.Background())
+	defer cancel()
+
+	iter, err := s.client.Cursor(ctx, &pb.CursorRequest{Namespace: wire.NamespaceVASPs})
+	if err != nil {
+		return err
+	}
+
+	for {
+		var pair *pb.KVPair
+		if pair, err = iter.Recv(); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+
+		vasp := &gds.VASP{}
+		if err = proto.Unmarshal(pair.Value, vasp); err != nil {
+			return err
+		}
+
+		names.Add(vasp.CommonName, vasp.Id)
+		for _, name := range vasp.Entity.Names() {
+			names.Add(name, vasp.Id)
+		}
+
+		websites.Add(vasp.Website, vasp.Id)
+		countries.Add(vasp.Entity.CountryOfRegistration, vasp.Id)
```

Comment:
We might want to also check the other two places that have country (Country of Issue and Address.Country). Country of Issue and Country of Registration are both option, while Address.Country is required.